rancher2/resource_rancher2_machine_config_v2.go

```diff
@@ -60,10 +60,6 @@ func waitForMachineConfigV2(d *schema.ResourceData, config *Config, interval tim
 			return nil
 		}
 		log.Printf("[INFO] Retrying on error Refreshing Machine Config V2 %s: %v", d.Id(), err)
-		if IsNotFound(err) || IsForbidden(err) {
```

Comment:
I don't believe the for loop is being used correctly in this function. If getMachineConfigV2ByID returns an error, the function will remain stuck in the loop indefinitely and never reach the select block. I’d recommend refactoring the logic to use a backoff function, similar to [OnError](https://pkg.go.dev/k8s.io/client-go/util/retry@v0.30.1#OnError), if you prefer not to introduce a new module.

Reply:
but the select block is inside the loop

Reply:
with this change there are only two return scenarios:
1. getMachineConfigV2ByID returns nil for the error (there wasn't an error)
2. the time is after the timeout and the context is done

am I missing something?

Reply:
I am not a fan of loops that don't end like this one, but without a good test suite (something I am working on) I am trying to reduce impact as much as possible.

Reply:
My bad. I misread the code and thought the select part was outside of the for loop when reviewing the code.  
But we still have the problem that the loop never ends: if `getMachineConfigV2ByID` somehow always returns a 404/403 error, then the function goes to the `case <-time.After(rancher2RetriesWait * time.Second):` case and keep retrying.  Right?

Reply:
I think we should unset the ID in the case of timing out, ie. adding `d.SetId("")` to the  `case <-ctx.Done():` section.

Reply:
done, thank you so much for your help!